File: software_creating/tasks/Task15.js
/* Вводится два числа М и N и массив размерностью M*N, найти индекс 
столбца в котором сумма элементов максимальна */

function findMaxColumn(arr) {
  let arrSize = arr.length;
  let rowSize = arr[0].length;
  let result = [];

  function sum(column) {
    let result = 0;
  
    for (let i = 0; i < arrSize; i++) {
      result += arr[i][column];
    }
  
    return result;
  }

  for (let i = 0; i < rowSize; i++) {
    result.push( sum(i) );
  }

  return result.indexOf( Math.max(...result) );
}

let myArr = [
  [3, 5, 9],
  [2, 7, 8],
  [1, 7, 3],
  [2, 0, 5]
]

console.log( findMaxColumn(myArr) );

File: software_creating/tasks/Task15.c
#include <stdio.h>

// Вводится два числа M и N и массив размерностью MxN
// Найти индекс столбца, в котором сумма элементов максимальна

int main() {
    int M, N;
    printf("M = ");
    scanf("%d", &M);
    printf("N = ");
    scanf("%d", &N);
    int arr[M][N];
    for (int i = 0; i < M; i++) {
        for (int j = 0; j < N; j++) {
            printf("arr[%d][%d] = ", i, j);
            scanf("%d", &arr[i][j]);
        }
    }
    int idx = 0, sum, max = 0;
    for (int i = 0; i < M; i++)
        max += arr[i][0];
    for (int j = 1; j < N; j++) {
        sum = 0;
        for (int i = 0; i < M; i++)
            sum += arr[i][j];
        if (sum >= max) {
            max = sum;
            idx = j;
        }
    }
    printf("%d", idx);
    return 0;
}
